src/queueUntilType.js

```diff
@@ -0,0 +1,35 @@
+import { subscribeToResult } from 'rxjs/util/subscribeToResult';
+
+export function queueUntilType(expectedType) {
+  return this.lift(function(source) {
+    let queue = [];
+    let hasSeenType = false;
+    const nullifyQueue = () => { queue = null; };
+
+    const outer = {
+      notifyNext: (_, action) => this.next(action),
+      notifyComplete: nullifyQueue
+    };
+    return source.subscribe({
+      next: action => {
+        // It's important to check this first because of possible re-entry
+        // before we've finished flushing the queue.
+        if (hasSeenType) {
+          return this.next(action);
+        }
+
+        if (action && action.type === expectedType) {
+          hasSeenType = true;
+          this.add(subscribeToResult(outer, queue));
+          // no longer needed, free up memory
+          nullifyQueue();
+        } else {
+          queue.push(action);
+        }
+      },
+      error: e => this.error(e),
+      complete: () => this.complete(),
+      unsubscribe: nullifyQueue
```

Comment:
Oh I missed your linking to the source. Hmm that does indeed suggest at first glance that it would be called..but I can't reproduce. This is obv secondary to this PR, I'm just indeed very curious! 😄